core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1062,16 +1062,18 @@ private void setTaskDomains(List<Task> tasks, Workflow wf) {
         }
     }
 
-    private String getActiveDomain(String taskType, String[] domains) {
+    @VisibleForTesting
+    String getActiveDomain(String taskType, String[] domains) {
         // The domain list has to be ordered.
-        // In sequence check if any worker has polled for last 30 seconds, if so that isSystemTask the Active domain
+        // In sequence check if any worker has polled for last 30 seconds, if so that is the Active domain
+
         return Arrays.stream(domains)
                 .map(domain -> executionDAO.getPollData(taskType, domain.trim()))
                 .filter(Objects::nonNull)
                 .filter(validateLastPolledTime)
                 .findFirst()
                 .map(PollData::getDomain)
-                .orElse(null);
+                .orElse(domains[domains.length - 1].trim());
```

Comment:
Would it be possible to throw ArrayIndexOutOfBoundsException when domains length is 0?

Reply:
As mentioned above, it is guaranteed that the domains array is atleast of size 1 when passed as param into the function. I will update the javadoc.

Reply:
Added checks and tests